sdk/nodejs/runtime/mocks.ts

```diff
@@ -206,8 +206,14 @@ export class MockMonitor {
     }
 
     public supportsFeature(req: any, callback: (err: any, innerResponse: any) => void) {
+        const id = req.getId();
+
+        // Support for "outputValues" is deliberately disabled for the mock monitor so
+        // instances of `Output` don't show up in `MockResourceArgs` inputs.
+        const hasSupport = id === "secrets" || id === "resourceReferences";
```

Comment:
Instead of blanket support for all features, I've changed it to be explicit about what the mock monitor supports. These are based off what the actual engine supports:

https://github.com/pulumi/pulumi/blob/b8b6f3dd70b2e35e15a4a5e69a9dadebf8be5932/pkg/resource/deploy/source_eval.go#L539-L546